### PR TITLE
feat!: Docker CI - Security added & Tagging updates

### PR DESCRIPTION
**Pull request by jrmanes**
## Overview

Hello team, 

Hope you’re doing well 

The following PR contains some changes in the Docker CI, build & publish.

I’ve added some features:
- File renamed: `amd64-docker-build.yml` -> `docker-build-publish.yml`
*We will be able to use a matrix for building the container in different architectures.*

- This CI is triggered when
  - When push to any branch
  - When creating `PRs`
  - When there's a new hotfix (when push to main branch)
  - When push tags `(v0.0.0/v0.0.0-alpha/v0.0.0-beta/v0.0.0-rcX)`

- Security: 
[Trivy](https://www.aquasec.com/products/trivy/) added to the CI, this is the first step during the build, it scans the image and provide us a table summary in case we have any CVE, if everything goes well, it continues with the next step.
*It’s not going to block the CI in case of bugs, though I think it will be good for us, stop it and fix them.*

- Docker tagging:
  - Git short SHA => for example: `a594b69`
  - Tag ID => for example: `0.0.1`
  - Always keep updated the latest tag
- Provided some additional metadata to the images:
  - Maintainer  =>   "maintainer": "CelestiaOrg"
  - Description => "CelestiaOrg repository celestiaorg/celestia-app"
  - URL to the specific commit =>  "commit_url": "https://github.com/celestiaorg/celestia-app/commit/a594b691ac019fc39afc7ee28c2798269ef0b159"
  - Docker pull command => "docker_pull_command": "docker pull ghcr.io/celestiaorg/celestia-app:a594b691"

---

## Checklist

- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [x] Linked issues closed with keywords

---

## Blockers

Hello team!
I'll need to add some permissions to allow the CI to have the access to the packages.

This is an error that I'm having: [link](https://github.com/celestiaorg/celestia-app/actions/runs/4058197292/jobs/6984898432)

```
ERROR: failed to solve: failed to push ghcr.io/celestiaorg/celestia-app:pr-1320: unexpected status: 403 Forbidden
Error: buildx failed with: ERROR: failed to solve: failed to push ghcr.io/celestiaorg/celestia-app:pr-1320: unexpected status: 403 Forbidden
```

Thanks in advance!

cc: @evan-forbes @rootulp  @Bidon15 @sysrex 

Please, ping me when you'll going to merge it, just to check that everything goes fine 😊 
Thank you team!

--- 
Closes Issue: [37](https://github.com/celestiaorg/devops/issues/37)

### CODE REVIEW

**Review by Bidon15 — APPROVED** [review]
LGTM

**Review by SweeXordious — COMMENTED** [review]
LGTM :+1: :+1: 

[Question]
Is there a reason why this workflow not triggered for this PR? It's a PR targeting main...

**Review by SweeXordious — APPROVED** [review]
Nvm, just checked the commit history.

Approving to merge once you have permissions and everything is green :smile_cat:

**Comment by jrmanes** [comment]
> LGTM 👍 👍
> 
> [Question] Is there a reason why this workflow not triggered for this PR? It's a PR targeting main...

hey @SweeXordious thanks for your review 😊 

If I'm not wrong, yes, it was triggered as a PR to `main` [here](https://github.com/celestiaorg/celestia-app/actions/workflows/docker-build-publish.yml)
There is where I found the permissions issue when trying to publish the new image: `ghcr.io/celestiaorg/celestia-app:pr-1320`

**Review by evan-forbes — APPROVED** [review]
LGTM, we should be able to merge after the conflicts and CI is completed

File: .github/workflows/docker-build-publish.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+    steps:
+      - name: Checkout
+        uses: "actions/checkout@v3"
+
+      - name: Login to GHCR
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
```
[the remark this answers is not in the document]
Comment (jrmanes) [reply .github/workflows/docker-build-publish.yml]: good catch!  it's possible that this was the issue, I'm gonna change it, thank you!

[the remark this answers is not in the document]
Comment (jrmanes) [reply .github/workflows/docker-build-publish.yml]: I've changed it in this [PR](https://github.com/celestiaorg/celestia-app/pull/1327), I had a problem and this PR was closed 🤦

**Pull request closed without merging**